src/app/clusters/scenes/SceneTable.h

```diff
@@ -0,0 +1,303 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/clusters/scenes/ExtensionFieldSets.h>
+#include <lib/support/CHIPMemString.h>
+#include <lib/support/CommonIterator.h>
+#include <lib/support/IntrusiveList.h>
+#include <lib/support/PersistentData.h>
+#include <lib/support/Span.h>
+
+namespace chip {
+namespace scenes {
+
+// Storage index for scenes in nvm
+typedef uint8_t SceneIndex;
+
+typedef uint32_t TransitionTimeMs;
+typedef uint32_t SceneTransitionTime;
+
+constexpr GroupId kGlobalGroupSceneId        = 0x0000;
+constexpr SceneIndex kUndefinedSceneIndex    = 0xff;
+constexpr SceneId kUndefinedSceneId          = 0xff;
+static constexpr uint8_t kMaxScenesPerFabric = CHIP_CONFIG_SCENES_MAX_PER_FABRIC;
+
+static constexpr size_t kIteratorsMax       = CHIP_CONFIG_MAX_SCENES_CONCURRENT_ITERATORS;
+static constexpr size_t kSceneNameMaxLength = CHIP_CONFIG_SCENES_CLUSTER_MAXIMUM_NAME_LENGTH;
+
+/// @brief SceneHandlers are meant as interface between various clusters and the Scene table.
+/// When a scene command involving extension field sets is received, the Scene Table will go through
+/// the list of handlers to either retrieve, populate or apply those extension field sets.
+///
+/// Generally, for each specific <endpoint, cluster> pair there should be one and only one handler
+/// registered with the scene table that claims to handle that pair.
+///
+/// A SceneHandler can handle a single <endpoint, cluster> pair, or many such pairs.
+///
+/// @note If more than one handler claims to handl a specific <endpoint, cluster> pair, only one of

```

Comment:
```suggestion
/// @note If more than one handler claims to handle a specific <endpoint, cluster> pair, only one of
```
This was my typo, sorry...